src/Hcal/HcalRecProducer.cxx

```diff
@@ -0,0 +1,206 @@
+        position.SetX(pos);
+      else
+        position.SetY(pos);
+
+      // time
+      hitTime = fabs(timeRelClock25_close + timeRelClock25_far) / 2;
+
+      // attenuation
+      double att_close =
+          exp(-1 * ((half_total_width - fabs(pos)) / 1000.) / attlength_);
+      double att_far =
+          exp(-1 * ((half_total_width + fabs(pos)) / 1000.) / attlength_);
+
+      if (digi_close.isTOT()) {
+        double voltage_close = (digi_close.tot() - pedestal_) * gain_;
+        double voltage_far = (digi_far.tot() - pedestal_) * gain_;
+        voltage =
+            (voltage_close / att_close + voltage_far / att_far) / 2;  // mV
+        voltage_min =
+            std::min(voltage_close / att_close, voltage_far / att_far);
+      } else {
+        double maxMeas_close{0.};
+        int numWholeClocks{0};
+        for (auto it = digi_close.begin(); it < digi_close.end(); it++) {
+          double amplitude = (it->adc_t() - pedestal_) * gain_;
+          double time = numWholeClocks * clock_cycle_;  // ns
+          if (amplitude > maxMeas_close) maxMeas_close = amplitude;
+        }
+        double maxMeas_far{0.};
+        numWholeClocks = 0;
+        for (auto it = digi_far.begin(); it < digi_far.end(); it++) {
+          double amplitude = (it->adc_t() - pedestal_) * gain_;
+          double time = numWholeClocks * clock_cycle_;  // ns
+          if (amplitude > maxMeas_far) maxMeas_far = amplitude;
+        }
+        voltage =
+            (maxMeas_close / att_close + maxMeas_far / att_far) / 2;  // mV
+        voltage_min =
+            std::min(maxMeas_close / att_close, maxMeas_far / att_far);
+      }
+
+      iDigi += 2;
+    }       // end double readout loop
+    else {  // single readout
+
+      if (digi.isTOT()) {
+        // TOT - number of clock ticks that pulse was over threshold
+        // this is related to the amplitude of the pulse approximately through a
+        // linear drain rate the amplitude of the pulse is related to the energy
+        // deposited
+
+        // convert the time over threshold into a total energy deposited in the
+        // bar (time over threshold [ns] - pedestal) * gain
+        voltage = (digi.tot() - pedestal_) * gain_;
+      } else {
+        // ADC mode of readout
+        // ADC - voltage measurement at a specific time of the pulse
+        TH1F measurements("measurements", "measurements", 10. * clock_cycle_,
+                          0., 10. * clock_cycle_);
+        double maxMeas{0.};
+        int numWholeClocks{0};
+        for (auto it = digi.begin(); it < digi.end(); it++) {
+          double amplitude_fC = (it->adc_t() - pedestal_) * gain_;
+          double time =
+              numWholeClocks * clock_cycle_;  //+ offestWithinClock; //ns
+          measurements.Fill(time, amplitude_fC);
+          if (amplitude_fC > maxMeas) maxMeas = amplitude_fC;
+        }
+        // just use the maximum measured voltage
+        voltage = maxMeas;
```

Comment:
If include attenuation in single-ended digi, you will need to account for that here as well.

Reply:
Ok so this is the only part of your comments where I run into conflicts. After having implemented the attenuation in the DigiProducer I realized that there is no way to take the attenuation back when reconstructing the pulse for the sideHcal because the coordinate along the bar is unknown.
For now it is taken always as the half width of the bar (as returned from the HcalGeometry class). 

This means that this `pos` variable will always equal the half width and reverting the attenuation will give the same result for all the hits in that section of the sideHcal (the attenuation reversed will be more precise for the cases where the simhit position along the bar is closest to the half width of the bar).
```
double att_close =
          exp(-1. * ((half_total_width - fabs(pos)) / 1000.) / attlength_);
double att_far =
          exp(-1. * ((half_total_width + fabs(pos)) / 1000.) / attlength_);
```
So not sure what the point is to include the attenuation if we cannot be somewhat precise in reversing it. @tomeichlersmith  do you have any suggestions?

Reply:
I think including the attenuation in a realistic way such that we _can't_ be precise in reversing it is **important**. This leads the digi/reco pipeline to more realistically model the accuracy we would actually be able to deliver and prevents us from thinking the Side HCal is more precise that it actually is.

I think your solution here of digitizing with the actual attenuation and the reconstructing with the attenuation assuming the hit was at the center of the bar is a good first approximation. I have no doubt there are fancier reconstruction methods that could be implemented, but again, this is already a lot of great work.